main/transaction/index.ts

```diff
@@ -1,170 +1,187 @@
-  const useFrameMaxPriorityFeePerGas = !rawTx.maxPriorityFeePerGas || isNaN(parseInt(rawTx.maxPriorityFeePerGas, 16))
-  
+  txData.type = intToHex(2);
+
+  const useFrameMaxFeePerGas =
+    !rawTx.maxFeePerGas || isNaN(parseInt(rawTx.maxFeePerGas, 16));
+  const useFrameMaxPriorityFeePerGas =
+    !rawTx.maxPriorityFeePerGas ||
+    isNaN(parseInt(rawTx.maxPriorityFeePerGas, 16));
+
   if (!useFrameMaxFeePerGas && !useFrameMaxPriorityFeePerGas) {
     // return tx unaltered when we are using no Frame-supplied values
-    return txData
+    return txData;
   }
 
   if (useFrameMaxFeePerGas && useFrameMaxPriorityFeePerGas) {
     // dapp did not supply a valid value for maxFeePerGas or maxPriorityFeePerGas so we change the source flag
-    txData.gasFeesSource = GasFeesSource.Frame
+    txData.gasFeesSource = GasFeesSource.Frame;
   }
 
-  const maxPriorityFee = useFrameMaxPriorityFeePerGas ? gas.price.fees.maxPriorityFeePerGas : rawTx.maxPriorityFeePerGas as string
+  const maxPriorityFee = useFrameMaxPriorityFeePerGas
+    ? gas.price.fees.maxPriorityFeePerGas
+    : (rawTx.maxPriorityFeePerGas as string);
 
   // if no valid dapp-supplied value for maxFeePerGas we calculate it
-  txData.maxFeePerGas = useFrameMaxFeePerGas ? calculateMaxFeePerGas(gas.price.fees.maxBaseFeePerGas, maxPriorityFee) : txData.maxFeePerGas
+  txData.maxFeePerGas = useFrameMaxFeePerGas
+    ? calculateMaxFeePerGas(gas.price.fees.maxBaseFeePerGas, maxPriorityFee)
+    : txData.maxFeePerGas;
 
   // if no valid dapp-supplied value for maxPriorityFeePerGas we use the Frame-supplied value
-  txData.maxPriorityFeePerGas = useFrameMaxPriorityFeePerGas ? bnToHex(toBN(maxPriorityFee)) : txData.maxPriorityFeePerGas
-  
-  return txData
+  txData.maxPriorityFeePerGas = useFrameMaxPriorityFeePerGas
+    ? addHexPrefix(BigNumber(maxPriorityFee).toString(16))
+    : txData.maxPriorityFeePerGas;
+
+  return txData;
 }
 
-function hexifySignature ({ v, r, s }: Signature) {
-  return {
-    v: addHexPrefix(v),
+function serializeSignature({ v, r, s }: Signature) {
+  return splitSignature({
+    v: hexToInt(v),
     r: addHexPrefix(r),
-    s: addHexPrefix(s)
-  }
+    s: addHexPrefix(s),
+  });
 }
 
-async function sign (rawTx: TransactionData, signingFn: (tx: TypedTransaction) => Promise<Signature>) {
-  const common = chainConfig(parseInt(rawTx.chainId), parseInt(rawTx.type) === 2 ? 'london' : 'berlin')
-
-  const tx = TransactionFactory.fromTxData(rawTx, { common })
-
-  return signingFn(tx).then(sig => {
-    const signature = hexifySignature(sig)
-
-    return TransactionFactory.fromTxData(
-      {
-      ...rawTx,
-      ...signature
-      },
-      { common }
-    )
-  })
+async function sign(
+  rawTx: TransactionData,
+  signingFn: (tx: TransactionData) => Promise<Signature>
+): Promise<string> {
+  return signingFn(rawTx).then((sig) => {
+    const signature = serializeSignature(sig);
+    const unsignedTx = convertToUnsignedTransaction(rawTx)
```

Comment:
we should probably make this call beforehand and pass the unsigned tx to the callback since I don't think any of the callback handlers need the `TransactionData` object